### PR TITLE
Added details to Map

### LINKED ISSUE #2437: Add details section for the map card and burger menu
### Description
The map needs a details section containing an editor for adding video, link, etc.

### Please indicate if this issue is related to a bug or a new feature request
- [x] New Feature

### In case of New Feature (otherwise remove this paragraph)
*Acceptance Criteria - AC*

- [ ] Managing and editing of the details section
- [ ] Add it to the card in the Home
- [ ] Add it to the burger menu in the MapViewer

### Other useful information (optional):
see the mockups [here](http://demo.geo-solutions.it/mockups/mapstore2/#/maps-properties)

### PR DESCRIPTION
**Pull request by MV88**
## Description
Added details in viewer page (under burger menu) if they are present for the opened map.
Added details section in map card
various actions available for details:
- create
- modify
- delete 
- show readonly from the card tools 

## Issues
 - Fix #2437

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
Actually is not possible to add details to the map and therefore to change/delete them

**What is the new behavior?**
it is possible from the homepage map card.

**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: 

**Other information**:
You need to add DETAILS category to the configured geostore

### CODE REVIEW

File: web/client/actions/maps.js [hunk 1]
```diff
@@ -640,19 +666,170 @@ function createMap(metadata, content, thumbnail, options) {
...
+function setDetailsChanged(detailsChanged) {
+    return {
+        type: SET_DETAILS_CHANGED,
+        detailsChanged
+    };
+}
+/**
+ * open details
+ * @memberof actions.maps
+ * @return {action}        type `OPEN_AND_FETCH_DETAILS`
```
Comment (offtherailz) [inline web/client/actions/maps.js]: keep it aligned

File: web/client/components/maps/MapCard.jsx [hunk 2]
```diff
@@ -74,25 +78,40 @@ class MapCard extends React.Component {
...
+                    this.stopPropagate(evt);
+                    this.onEdit(this.props.map, false);
+                    if (!this.props.map.detailsText) {
+                        this.props.detailsSheetActions.onOpenOrFetchDetails({open: true, fetch: true, readOnly: true});
+                    } else {
+                        this.props.detailsSheetActions.onToggleDetailsSheet(true);
+                    }
+                },
+                glyph: "chevron-up",
+                tooltip: LocaleUtils.getMessageById(this.context.messages, "map.details.show")
```
Comment (offtherailz) [inline web/client/components/maps/MapCard.jsx]: change to sheet icon

File: web/client/components/maps/MapCard.jsx [hunk 3]
```diff
@@ -74,25 +78,40 @@ class MapCard extends React.Component {
...
-                onClick: (evt) => {this.stopPropagate(evt); this.displayDeleteDialog(); },
-                glyph: "remove-circle",
-                disabled: this.props.map.deleting,
-                loading: this.props.map.deleting,
-                tooltip: LocaleUtils.getMessageById(this.context.messages, "manager.deleteMap")
+                onClick: (evt) => {
+                    this.stopPropagate(evt);
+                    this.onEdit(this.props.map, false);
+                    if (!this.props.map.detailsText) {
+                        this.props.detailsSheetActions.onOpenOrFetchDetails({open: true, fetch: true, readOnly: true});
```
Comment (offtherailz) [inline web/client/components/maps/MapCard.jsx]: trigger fetch (if details are not present) with an epic removing it from UI part. 
UI should only show the detail text or loading if present but not loaded yet. Pay attention to empty strings

**Comment by MV88** [comment]
Wating for #2505 to be merged before update this

**Comment by offtherailz** [comment]
This conflicts, please resolve conflicts

**Comment by MV88** [comment]
if you see my comment i'm waiting another [PR](https://github.com/geosolutions-it/MapStore2/pull/2505) to be merged before i do any fix on this.

**Comment by offtherailz** [comment]
Please solve conflicts, we can move on, #2505 is merged

**Comment by MV88** [comment]
closing this, solving conflicts [here](https://github.com/geosolutions-it/MapStore2/pull/2528)

**Pull request closed without merging**